tools/bouncer_worker/src/lib/config.js

```diff
@@ -18,57 +18,101 @@
 const path = require('path');
 const fs = require('fs');
 const { v4: uuidv4 } = require('uuid');
+const { object, string, number } = require('yup');
 const { exitApplication } = require('./utils');
 const params = require('./processParams');
 
 const Config = {};
 
-/* eslint-disable no-param-reassign */
-const applyDefaultValuesIfUndefined = (config) => {
-	config.timeoutMS = config.timeoutMS || 180 * 60 * 1000;
-
-	// rabbitmq configurations
-	config.rabbitmq.maxRetries = config.rabbitmq.maxRetries || 3;
-	config.rabbitmq.task_prefetch = config.rabbitmq.task_prefetch || 4;
-	config.rabbitmq.model_prefetch = config.rabbitmq.model_prefetch || 1;
-	config.rabbitmq.drawing_prefetch = config.rabbitmq.drawing_prefetch || 1;
-	config.rabbitmq.pollingIntervalMS = config.rabbitmq.pollingIntervalMS || 10 * 1000;
-	config.rabbitmq.maxWaitTimeMS = config.rabbitmq.maxWaitTimeMS || 5 * 60 * 1000;
-	config.rabbitmq.waitBeforeShutdownMS = config.rabbitmq.waitBeforeShutdownMS || 60000;
-
-	// logging related
-	config.logging = config.logging || {};
-	config.logging.taskLogDir = config.logging.taskLogDir || config.bouncer.log_dir || config.rabbitmq.sharedDir;
-	config.logging.workerLogPath = config.logging.workerLogPath || config.logLocation;
-	config.logging.logLevel = config.logging.logLevel || 'info';
-
-	// processMonitoring
-	config.processMonitoring = config.processMonitoring || {};
-	config.processMonitoring.memoryIntervalMS = config.processMonitoring.memoryIntervalMS || 100;
-
-	// create connection string for db
-	if (!config.db.connectionString) {
-		config.db.connectionString = `mongodb://${config.db.dbhost}:${config.db.dbport}`;
-	}
+const rabbitmq = object({
+	host: string().required(),
+	sharedDir: string().required(),
+	worker_queue: string(),
+	callback_queue: string(),
+	model_queue: string(),
+	drawing_queue: string(),
+	task_prefetch: number().default(4),
+	model_prefetch: number().default(1),
+	drawing_prefetch: number().default(1),
+	pollingIntervalMS: number().default(10 * 1000),
+	maxRetries: number().default(3),
+	waitBeforeShutdownMS: number().default(60000),
+	maxWaitTimeMS: number().default(5 * 60 * 1000),
+});
 
-	if (config.repoLicense) {
-		config.instanceId = uuidv4();
-	}
-};
-/* eslint-enable no-param-reassign */
+const bouncer = object({
+	path: string().required(),
+	envars: object(),
+	log_dir: string(),
+});
+
+const logging = object({
+	taskLogDir: string(),
+	workerLogPath: string(),
+	logLevel: string().oneOf([ // Options are the six winston log levels https://www.npmjs.com/package/winston/v/2.4.6#logging-levels
+		'error',
+		'warn',
+		'info',
+		'verbose',
+		'debug',
+		'silly',
+	]).default('error'),
+});
+
+const processMonitoring = object({
+	memoryIntervalMS: number().default(100),
+});
+
+const db = object({
+	dbhost: string(),
+	dbport: number().min(1).max(65536),
+});
+
+const schema = object({
+	timeoutMS: number().default(180 * 60 * 1000),
+	logLocation: string(),
```

Comment:
I think logLocation is just old place where we put workerLogPath?

I think we can probably get rid of it